src/js/stores/NewsFeedStore.jsx

```diff
@@ -187,13 +295,32 @@ class NewsFeedStore extends EventEmitter {
         this.emit('updated');
         break;
       }
+      case 'FLAGGED_GET': {
+        this.dataFetched = true;
+        Array.prototype.push.apply(this.flaggedPosts.items, action.response.data.posts[0]);
+        this.contentFeed.nextOffset = action.response.data.nextOffset;
+        this.contentFeed.count = action.response.data.count;
+        this.emit('updated');
+        break;
+      }
       case 'NEWSFEED_FROM_STORE': {
         // Data is already in the store
         this.emit('updated');
         break;
       }
       case 'CREATE_POST_SUCCESS': {
+        let newPost = action.post;
+        newPost.current_user_likes = false;
+        newPost.like_count = 0;
+        newPost.timestamp = newPost.id;
+        this.contentFeed.items.unshift(newPost);
         this.emit('post');
+        if (newPost.media_link) {
+          // Photos take time to load
+          setTimeout(function(){newsFeedStore.emit('updated')}, 1800);
```

Comment:
is there no way this could have been wrapped in a callback instead of just waiting a random amount of time.